lib/skate_web/channels/channel_auth.ex

```diff
@@ -6,6 +6,12 @@ defmodule SkateWeb.ChannelAuth do
   """
   @spec valid_token?(Phoenix.Socket.t()) :: boolean()
   def valid_token?(socket) do
+    token_fn = Application.get_env(:skate, :valid_token_fn, &_valid_token?/1)
+    token_fn.(socket)
+  end
+
+  @spec _valid_token?(Phoenix.Socket.t()) :: boolean()
+  defp _valid_token?(socket) do
```

Comment:
**nitpick:** I know in Elixir, prefixing a variable with `_` causes the compiler to not throw a warning if it isn't used. I'm trying to make up my mind as to whether using it in this context is confusing or not, but I also don't have a better idea. Maybe `default_valid_token?` or something?

Reply:
Oh right, I don't wanna keep the leading underscore, I'll think on it!

Reply:
@lemald what do you think of `valid_socket_token?`?

Reply:
Sure, that works.

Reply:
interesting note
https://hexdocs.pm/elixir/writing-documentation.html#hiding-internal-modules-and-functions
> Start the function name with one or two underscores, for example, `__add__/2`. Functions starting with underscore are automatically treated as hidden, although you can also be explicit and add `@doc false`. The compiler does not import functions with leading underscores and they hint to anyone reading the code of their intended private usage.

Reply:
Interesting - usually we just use `defp` for private functions and leave it at that, but it's good to know that at least some parts of the Elixir documentation recommend the underscore approach.